### PR TITLE
msj/stanford-repogroup

### PR DESCRIPTION
**Pull request by RedHeadedStranger99**
Create sourcegraph.com/stanford page.

### CODE REVIEW

File: web/src/search/input/SearchPage.tsx [hunk 1]
```diff
@@ -90,7 +90,7 @@ export const SearchPage: React.FunctionComponent<Props> = props => {
             [codeInsightsEnabled, props.extensionsController.services.view]
         )
     )
-
+    console.log('###', props)
```
Comment (christinaforney) [inline web/src/search/input/SearchPage.tsx:93]: Oops :) Sneaky console.log.

File: web/src/repogroups/Stanford.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import { RepogroupMetadata } from './types'
+import { SearchPatternType } from '../../../shared/src/graphql/schema'
+import * as React from 'react'
+
+export const stanford: RepogroupMetadata = {
+    title: 'Stanford University',
+    name: 'stanford',
+    url: '/stanford',
+    description: 'Explore open-source code from Stanford students, faculty, research groups, and clubs.',
```
Comment (christinaforney) [inline web/src/repogroups/Stanford.tsx:9]: @kghopson - can you please help review the copy included in this file? The lines to look at are descriptions of searches, look at "description" and "title". We can also add longer descriptions to each of the individual examples if you have thoughts on that! LMK if you have questions.

File: web/src/repogroups/Stanford.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                <>
+                    <span className="repogroup-page__keyword-text">repo:</span>/RedHeadedStranger99/{' '}
+                </>
+            ),
+            patternType: SearchPatternType.literal,
+            rawQuery: 'repo:/RedHeadedStranger99/',
+        },
+    ],
+    homepageDescription: 'Explore Stanford open-source code.',
+    homepageIcon:
```
Comment (christinaforney) [inline web/src/repogroups/Stanford.tsx:75]: @rrhyne - can you please double check this logo?

File: enterprise/dev/Procfile [hunk 4]
```diff
@@ -5,7 +5,7 @@ searcher: searcher
 replacer: replacer
 symbols: symbols
 github-proxy: github-proxy
-frontend: env CONFIGURATION_MODE=server SITE_CONFIG_ESCAPE_HATCH_PATH=$HOME/.sourcegraph/site-config.json frontend
+frontend: env SOURCEGRAPHDOTCOM_MODE=TRUE CONFIGURATION_MODE=server SITE_CONFIG_ESCAPE_HATCH_PATH=$HOME/.sourcegraph/site-config.json frontend
```
Comment (christinaforney) [inline enterprise/dev/Procfile:8]: Is this something we want to add, or was this just temporary when you were testing?

File: dev/global-settings.json [hunk 5]
```diff
@@ -1,4 +1,6 @@
 {
   // This is set from an environment variable
-  "experimentalFeatures": {}
+  "experimentalFeatures": {
```
Comment (christinaforney) [inline dev/global-settings.json:3]: Do we want to turn this on for everyone?

File: web/src/repogroups/Stanford.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
+import { RepogroupMetadata } from './types'
+import { SearchPatternType } from '../../../shared/src/graphql/schema'
+import * as React from 'react'
+
+export const stanford: RepogroupMetadata = {
+    title: 'Stanford University',
+    name: 'stanford',
```
Comment (beyang) [inline web/src/repogroups/Stanford.tsx:7]: Does a "stanford" repogroup exist on sourcegraph.com yet? If not, @RedHeadedStranger99 you should add one by updating the global settings for sourcegraph.com. Note that the global settings on sourcegraph.com are not edited via the GUI, but rather by modifying the `sourcegraph-frontend.ConfigMap.yaml` file in the deploy-sourcegraph-dot-com repository. Ping me on Slack if anything's confusing here.

File: web/src/repogroups/Stanford.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    description: 'Explore open-source code from Stanford students, faculty, research groups, and clubs.',
+    examples: [
+        {
+            title: 'Explore Stanford machine learning projects.',
+            exampleQuery: <>machine learning</>,
+            patternType: SearchPatternType.literal,
+            rawQuery: 'machine learning',
+        },
+        {
+            title: 'Explore the code of specific research groups.',
```
Comment (beyang) [inline web/src/repogroups/Stanford.tsx:18]:
```suggestion
            title: 'Explore the code of specific research groups like Hazy Research, a group led by Professor Chris Ré that investigates weakly supervised machine learning models and the use of systems principles to automate the creation of training datasets.',
```

File: web/src/repogroups/Stanford.tsx [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+            exampleQuery: (
+                <>
+                    <span className="repogroup-page__keyword-text">repo:</span>/susom/{' '}
+                </>
+            ),
+            patternType: SearchPatternType.literal,
+            rawQuery: 'repo:/susom/',
+        },
+        {
+            title: 'Search for repositories related to introductory programming concepts.',
```
Comment (beyang) [inline web/src/repogroups/Stanford.tsx:38]: How about something like "Search through the source code the libraries provided to students in introductory computer science courses"? Is that expressible in a query (e.g,. `repogroup:stanford cs106a`) or maybe we need to create another repogroup for this?

File: web/src/repogroups/Stanford.tsx [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+            exampleQuery: (
+                <>
+                    <span className="repogroup-page__keyword-text">repo:</span>recursion{' '}
+                </>
+            ),
+            patternType: SearchPatternType.literal,
+            rawQuery: 'repo:recursion',
+        },
+        {
+            title: 'Explore the README files of thousands of projects.',
```
Comment (beyang) [inline web/src/repogroups/Stanford.tsx:48]: I don't think this example is compelling on its own. What might be interesting is searching for the mention of a particular string or regex pattern in all READMEs (and then we want to not just cover `README.txt`s but also `README`s and `README.md`s). What might be interesting is to combine this with the "machine learning" example, so you search for all READMEs that mention "machine learning". That seems like a pretty good proxy for "find all the machine learning projects at Stanford".

File: web/src/repogroups/Stanford.tsx [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+            exampleQuery: (
+                <>
+                    <span className="repogroup-page__keyword-text">file:</span>README.txt{' '}
+                </>
+            ),
+            patternType: SearchPatternType.literal,
+            rawQuery: 'file:README.txt',
+        },
+        {
+            title: "Perform a general search if you're unsure of an organization or project name.",
```
Comment (beyang) [inline web/src/repogroups/Stanford.tsx:58]: I'd delete these last two items.

File: web/src/repogroups/Stanford.tsx [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+        },
+        {
+            title: 'Explore the code of a specific user.',
+            exampleQuery: (
+                <>
+                    <span className="repogroup-page__keyword-text">repo:</span>/RedHeadedStranger99/{' '}
+                </>
+            ),
+            patternType: SearchPatternType.literal,
+            rawQuery: 'repo:/RedHeadedStranger99/',
```
Comment (beyang) [inline web/src/repogroups/Stanford.tsx:71]: This is technically not specific to searching over code belonging to a single user, but a single GitHub namespace, which could be an org or a user. Therefore, this is not really a distinct example and should be merged with the previous example that searches over code belonging to a particular organization.

File: web/src/repogroups/Stanford.tsx [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+        },
```
Comment (beyang) [inline web/src/repogroups/Stanford.tsx:72]: Can we add one regex search example, and one [structural search](https://docs.sourcegraph.com/user/search/structural) example, too?

File: web/src/repogroups/Stanford.tsx [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+                <>
+                    <span className="repogroup-page__keyword-text">file:</span>README machine learning{' '}
+                </>
+            ),
+            patternType: SearchPatternType.literal,
+            rawQuery: 'machine learning',
+        },
+        {
+            title:
+                'Explore the code of specific research groups like Hazy Research, a group led by Professor Chris Ré that investigates weakly supervised machine learning models and the use of systems principles to automate the creation of training datasets.',
```
Comment (beyang) [inline web/src/repogroups/Stanford.tsx:23]:
```suggestion
                'Explore the code of specific research groups like Hazy Research, a group that investigates machine learning models and automated training set creation.',
```

**Pull request merged**